Sources/tinys3/model/Multipart/MultipartUploadFile.swift

```diff
@@ -5,29 +5,17 @@ actor MultipartUploadFile {
     let fileSize: Int
     let partSize: Int
 
-    init(path: URL, partSize: Int? = nil) throws {
+    init(path: URL, partSize: Int? = nil, fileSize: Int? = nil) throws {
         self.handle = try FileHandle(forReadingFrom: path)
 
-        let fileSize = try FileManager.default.fileSize(of: path)
-
-        self.fileSize = fileSize
-        self.partSize = PartSizeCalculator.calculate(basedOn: fileSize)
+        let fileSizeUsed = try fileSize ?? FileManager.default.fileSize(of: path)
+        self.fileSize = fileSizeUsed
+        self.partSize = partSize ?? PartSizeCalculator.calculate(basedOn: fileSizeUsed)
     }
 
     var parts: [Range<Int>] {
-        var rangeStart = 0
-        var rangeEnd = -1
-
-        var parts = [Range<Int>]()
-
-        while rangeEnd < fileSize {
-            rangeStart = (parts.last?.upperBound ?? -1) + 1
-            rangeEnd = rangeStart + partSize
```

Comment:
The first seed of the off-by-one error: each next range started 1 after the end of the previous one… but those are open `Range` not `ClosedRange`, so the end of each range is not included